dvc/api/metrics.py

```diff
@@ -0,0 +1,162 @@
+from collections import Counter
+from typing import Dict, Optional
+
+from dvc.exceptions import DvcException
+from dvc.repo import Repo
+
+
+def metrics_show(
+    *targets: str,
+    all_branches: Optional[bool] = False,
+    all_tags: Optional[bool] = False,
+    all_commits: Optional[bool] = False,
```

Comment:
I don't have a strong opinion here, but I think I would prefer the Python API to only support a single `rev` argument and drop these arguments, even though it would be inconsistent with the CLI.

My reasoning is that these flags are convenient in CLI but in Python API would be trivial for the user to collect all the revisions, call `metrics_show(rev=rev)` in a loop, and store the results in the structure they find more convenient.

Reply:
Thanks, I like that better as well.  I couldn't reuse postprocess because with the multiple returns in the current version the metrics implementation had to be different.  When we simplify and drop all those args then it's much easier to bring together params and metrics.  I had started going this way but wanted to stay closer to cli so kept all the other args in there.

I'll make the changes and add tests next.  I appreciate the quick feedback!